diff --git a/twrp-functions.cpp b/twrp-functions.cpp
index 40205c9..f328eff 100644
--- a/twrp-functions.cpp
+++ b/twrp-functions.cpp
@@ -922,8 +922,8 @@ void TWFunc::Fixup_Time_On_Boot(const string& time_paths /* = "" */)
 
 	std::vector<std::string> paths; // space separated list of paths
 	if (time_paths.empty()) {
-		paths = Split_String("/data/system/time/ /data/time/", " ");
-		if (!PartitionManager.Mount_By_Path("/data", false))
+		paths = Split_String("/persist/time/ /data/vendor/time/", " ");
+		if (!PartitionManager.Mount_By_Path("/persist", false) && !PartitionManager.Mount_By_Path("/data", false))
 			return;
 	} else {
 		// When specific path(s) are used, Fixup_Time needs those
